### ISSUE TITLE
Google Drive Downloads not saving with file extension when using `backend copyid`

### ISSUE THREAD

**Issue by mikfreedman** [rclone/rclone#5196]
#### What is the problem you are having with rclone?
When using `rclone backend copyid` and downloading a google sheet document, I expect the filename to have the `.xlsx` extension.

#### What is your rclone version (output from `rclone version`)
```
rclone v1.56.0-beta.5356.9e2fbe0f1
- os/type: darwin
- os/arch: amd64
- go/version: go1.16.2
- go/linking: dynamic
- go/tags: cmount
```
#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
MacOS 10.15.7 (19H524) amd64

####  Which cloud storage system are you using? (eg Google Drive)
Google Drive

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)  

```
./rclone -vv  backend copyid drive: [drive-id] /tmp/test/
```

#### The rclone config contents with secrets removed.  

```
[drive]
type = drive
client_id = [client-id]
client_secret = [client-secret]
scope = drive
token = {"access_token":"[snipped]","expiry":"2021-04-02T18:06:55.213361-04:00"}

```

#### A log from the command with the `-vv` flag  

```
~/Downloads/rclone-v1.56.0-beta.5356.9e2fbe0f1-osx-amd64
➜ ./rclone -vv  backend copyid drive: [drive-id] /tmp/test/
2021/04/02 17:14:09 DEBUG : Using config file from "/Users/[user]/.config/rclone/rclone.conf"
2021/04/02 17:14:09 DEBUG : rclone: Version "v1.56.0-beta.5356.9e2fbe0f1" starting with parameters ["./rclone" "-vv" "backend" "copyid" "drive:" "[drive-id]" "/tmp/test/"]
2021/04/02 17:14:10 DEBUG : Google drive root '': root_folder_id = "[folder-id]" - save this in the config to speed up startup
2021/04/02 17:14:10 DEBUG : Creating backend with remote "/tmp/test/"
2021/04/02 17:14:12 DEBUG : Local file system at /tmp/test/: File to upload is small (4589 bytes), uploading instead of streaming
2021/04/02 17:14:12 DEBUG : temp-sheet: MD5 = f41f56705773e82b7e1c89f1c17f9f48 OK
2021/04/02 17:14:12 INFO  : temp-sheet: Copied (new)
2021/04/02 17:14:12 DEBUG : temp-sheet.xlsx: Updating size of doc after download to 4589
2021/04/02 17:14:12 INFO  : temp-sheet.xlsx: Copied (Rcat, new) to: temp-sheet
2021/04/02 17:14:12 DEBUG : 10 go routines active

~/Downloads/rclone-v1.56.0-beta.5356.9e2fbe0f1-osx-amd64 took 3s
➜ cd /tmp/test

/tmp/test
➜ ls
temp-sheet
```

You can see from the log that it downloads it as an xlsx, indeed if you change the filename it opens correctly in excel, however it does not update the local filename. This behavior is different from the regular `copy` command which correctly appends `.xlsx`

Copying a file from drive that already has an extension (e.g. `temp.csv`) on drive works as expected.

**Comment by mikfreedman** [comment]
FWIW I attempted to log this issue on the forum first but was told that new users cannot have posts with more than two links!

![image](https://user-images.githubusercontent.com/1176640/113455052-ab87fd80-93d7-11eb-8158-6e75e79976da.png)

**Comment by ncw** [comment]
This does indeed look like a bug.

Try this fix

[v1.56.0-beta.5357.951d314da.fix-5196-drive-copyid](https://beta.rclone.org/branch/fix-5196-drive-copyid/v1.56.0-beta.5357.951d314da.fix-5196-drive-copyid/) on branch [fix-5196-drive-copyid](https://github.com/rclone/rclone/tree/fix-5196-drive-copyid) (uploaded in 15-30 mins)

**Comment by mikfreedman** [comment]
```bash
➜ ./rclone -vv backend copyid drive: 1g4L2twAEOtUJe0zRERVUuabRjnJ-UXYqeifSb30L5v8 /tmp/test/
2021/04/03 10:42:29 DEBUG : Using config file from "/Users/[user]/.config/rclone/rclone.conf"
2021/04/03 10:42:29 DEBUG : rclone: Version "v1.56.0-beta.5357.951d314da.fix-5196-drive-copyid" starting with parameters ["./rclone" "-vv" "backend" "copyid" "drive:" "1g4L2twAEOtUJe0zRERVUuabRjnJ-UXYqeifSb30L5v8" "/tmp/test/"]
2021/04/03 10:42:29 DEBUG : Google drive root '': root_folder_id = "0ACDkaCghVYoEUk9PVA" - save this in the config to speed up startup
2021/04/03 10:42:29 DEBUG : Creating backend with remote "/tmp/test/"
2021/04/03 10:42:30 DEBUG : Local file system at /tmp/test/: File to upload is small (4683 bytes), uploading instead of streaming
2021/04/03 10:42:30 DEBUG : test-sheet.xlsx: MD5 = 5f819c36867ec7639f741a1267140281 OK
2021/04/03 10:42:30 INFO  : test-sheet.xlsx: Copied (new)
2021/04/03 10:42:30 DEBUG : test-sheet.xlsx: Updating size of doc after download to 4683
2021/04/03 10:42:30 INFO  : test-sheet.xlsx: Copied (Rcat, new)
2021/04/03 10:42:30 DEBUG : 10 go routines active

~/Downloads/rclone-v1.56.0-beta.5357.951d314da.fix-5196-drive-copyid-osx-arm64
➜ ls /tmp/test
test-sheet.xlsx
```

Confirmed working!

**Comment by mikfreedman** [comment]
@

**Comment by ncw** [comment]
Thanks for testing.

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.56.

I'll stick it in v1.55.1 too
